internal/pkg/cli/env_deploy.go

```diff
@@ -123,6 +136,15 @@ func (o *deployEnvOpts) Execute() error {
 	if err != nil {
 		return err
 	}
+	if mft.CDNConfig.CDNEnabled() && mft.HTTPConfig.Public.Certificates == nil && o.targetApp.Domain != "" {
+		describer, err := o.newEnvDescriber()
+		if err != nil {
+			return fmt.Errorf("describe env: %w", err)
+		}
+		if err := describer.ValidateCFServiceDomainAliases(); err != nil {
```

Comment:
What do you think of renaming this method like this?
I think it would also be super helpful to drop a comment why this is happening as it's not immediately obvious.
```go
if err := env.ValidateWebServicesHaveAliases(); err != nil {
   // With managed domain, if the customer isn't using `alias` the A-records are inserted in the service stack as each service domain is unique.
   // However, when clients enable CloudFront, they would need to update all their existing records to now point to the distribution. 
   // Hence, we force users to use `alias` and let the records be written in the environment stack instead.
   return err
}
```